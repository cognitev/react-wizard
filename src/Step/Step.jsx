/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Wizard Step wrapper
 * @param children
 * @param id
 */
const Step = ({ children, id }) => (
  <div
    id={id}
  >
    {children}
  </div>
);


Step.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  nextBtnTitle: PropTypes.string,
  prevBtnTitle: PropTypes.string,
  onEnter: PropTypes.func,
  onLeave: PropTypes.func,
  enabled: PropTypes.bool,
};

Step.defaultProps = {
  children: null,
  name: null,
  nextBtnTitle: null,
  prevBtnTitle: null,
  onEnter: null,
  onLeave: null,
  enabled: true,
};

export default Step;
