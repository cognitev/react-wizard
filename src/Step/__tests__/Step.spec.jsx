import React from 'react';
import { shallow } from 'enzyme';
import Step from '../Step';

describe('<Step /> Component', () => {
  it('should render children', () => {
    const rendered = shallow(
      <Step id="testStep">
        <div />
      </Step>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
