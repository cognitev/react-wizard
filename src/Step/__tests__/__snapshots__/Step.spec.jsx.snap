// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> Component should render children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Step
    enabled={true}
    id="testStep"
    name={null}
    nextBtnTitle={null}
    onEnter={null}
    onLeave={null}
    prevBtnTitle={null}
  >
    <div />
  </Step>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div />,
      "id": "testStep",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div />,
        "id": "testStep",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
